index.js

```diff
@@ -38,18 +38,32 @@ var defer = typeof setImmediate === 'function'
  *
  * @param {object} msg
  * @param {function} listener
+ * @param {boolean} lifo
  * @return {object}
  * @public
  */
 
-function onFinished (msg, listener) {
+function onFinished (msg, listener, lifo) {
   if (isFinished(msg) !== false) {
-    defer(listener, null, msg)
+    var attached = msg.__onFinished
```

Comment:
I'd love your thoughts on whether this might be a breaking change in semantics whereby all callbacks are run in the *same* future tick vs. each being possibly scheduled in *different* future ticks.